src/components/dumb/FilterInput.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div class="vl-grid filters-input">
+    <span v-if="!props?.options.length" class="vl-alert--warning">Geen filteropties geconfigureerd</span>
+    <template v-else>
+      <vl-select v-model:value="selectedOption" class="vl-col--5-12" mod-block mod-inline @update:value="clearInputs">
+        <option v-for="option in props.options" :key="option.key" :value="option">
+          {{ option.label }}
+        </option>
+      </vl-select>
+      <vl-input-group class="vl-col--7-12">
+        <slot
+          :value="filterInputValue?.value"
+          :set-value="setFilterInputValue"
+          :selected-option="selectedOption"
+        ></slot>
+
+        <vl-input-addon
+          :mod-disabled="filterValuesAreEmpty"
+          :disabled="filterValuesAreEmpty"
+          tag-name="button"
+          type="button"
+          icon="plus"
+          tooltip="Filter toevoegen"
+          text="Filter toevoegen"
+          @click="addFilter"
+        />
+      </vl-input-group>
+    </template>
+  </div>
+  <div v-if="!!filters.length" class="vl-grid filters-selected">
+    <span class="vl-col--1-12">Filters:</span>
+    <vl-action-group class="vl-col--10-12">
+      <vl-pill class="vl-u-spacer-left--xsmall vl-u-spacer-bottom--xsmall" mod-clickable @click="filters = []"
+        >Alle filters wissen</vl-pill
+      >
+      <vl-pill
+        v-for="filter in filters"
+        :key="filter.key"
+        mod-closable
+        class="vl-u-spacer-left--xsmall vl-u-spacer-bottom--xsmall"
+        @close="removeFilter(filter)"
+      >
+        {{ filter.label }} / {{ filter.value.label }}
+      </vl-pill>
+    </vl-action-group>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { VlInputGroup, VlInputAddon, VlActionGroup, VlSelect, VlPill } from '@govflanders/vl-ui-design-system-vue3';
+import type { IFilter, IFilterInputProps, IFilterOption } from '@models/filter-input';
+import { isEmpty, remove } from 'lodash';
+import { computed, ref, watch } from 'vue';
+
+const props = withDefaults(defineProps<IFilterInputProps>(), {
+  options: () => [],
+});
+const emit = defineEmits<{
+  (e: 'filters-selected', filters: IFilter[]): void;
+}>();
+
+// Filter option
+const selectedOption = ref<IFilterOption>(props.options[0]);
+
+// Filter input
+const filterInputValue = ref<{ value?: unknown; label?: string }>();
```

Comment:
Ik zou hier proberen die unknown weg te werken. Voor zover ik zie gaat dat string | IGemeente zijn? Dan weten we toch al wat we binnenkrijgen hier

Reply:
Momenteel is dat idd string of IGemeente, maar het gaat ook provincies, aanduidingsobjecten etc moeten ondersteunen, en dan is er telkens een change nodig en is het niet meer generiek. 
Ga eens kijken of dit op de een of andere manier met generics zou kunnen werken.

Reply:
Deze weet ik niet meteen hoe te fixen.
Generics is gelimiteerd op props (https://vuejs.org/api/sfc-script-setup.html#generics).
Via slots krijg ik dit niet geregeld. Dus voorlopig zie ik enkel die `unknown` als oplossing om het generiek te houden.

Reply:
en zo? 
![image](https://github.com/OnroerendErfgoed/vue_component_library/assets/9083121/5b0f8945-16c6-40e2-8d50-57887dcc8bab)

![image](https://github.com/OnroerendErfgoed/vue_component_library/assets/9083121/b96eebef-d16d-481f-ad07-ac836758cd63)

Dan moeten we dat inderdaad elke keer gaan uitbreiden met een nieuw type, maar liever dat dan na zoveel weken niet meer juist weten wat dat allemaal kan zijn en manueel moeten gaan zoeken